src/queries/queries.services.ts

```diff
@@ -101,6 +121,117 @@ export function removeQuery(
   );
 }
 
+export function findQueryJob(
+  ctx: AppBindings,
+  _id: UUID,
+): E.Effect<
+  QueryJobDocument,
+  DocumentNotFoundError | PrimaryCollectionNotFoundError | DatabaseError
+> {
+  return pipe(QueriesJobsRepository.findOne(ctx, { _id }));
+}
+
+export function addQueryJob(
+  ctx: AppBindings,
+  queryId: UUID,
+): E.Effect<
+  InsertOneResult<QueryJobDocument>,
+  DocumentNotFoundError | PrimaryCollectionNotFoundError | DatabaseError
+> {
+  const document = QueriesJobsRepository.toQueryJobDocument(queryId);
+  return QueriesJobsRepository.insert(ctx, document);
+}
+
+export function updateQueryJob(
+  ctx: AppBindings,
+  payload: {
+    jobId: UUID;
+    status: QueryJobStatus;
+    startedAt?: Date;
+    endedAt?: Date;
+    result?: JsonValue;
+    error?:
+      | DocumentNotFoundError
+      | PrimaryCollectionNotFoundError
+      | DatabaseError
+      | DataValidationError
+      | VariableInjectionError
+      | DataCollectionNotFoundError
+      | TimeoutError;
+  },
```

Comment:
Given this is a fairly complex type, maybe move it into a defined type? Eg `type Payload = { ... }`, rather than inline?